enhancements/sig-architecture/58-addon-configuration/README.md

```diff
@@ -0,0 +1,592 @@
+    // +optional
+    CRDName string `json:"crdName"`
+
+    // crName is the name of the CR used to configure instances of the managed add-on.
+    // This field should be configured if add-on CR have a consistent name across the all of the ManagedCluster
+    // instaces.
+    // +optional
+    CRName string `json:"crName"`
+
+    // lastObservedGeneration is the observed generation of the custom resource for the configuration of the add-on.
+    // +optional
+    LastObservedGeneration int64 `json:"lastObservedGeneration,omitempty"`
+}
+```
+
+But the `ConfigCoordinates` has some limits, for example, it only supports to use `CustomResourceDefinition` as the 
+add-on configuration, and it's not easy to locate the configuration object for add-on implementation.
+
+To make the `ConfigCoordinates` more flexible, we want to refactor the `ConfigCoordinates` to
+
+``` golang
+// ConfigCoordinates represents the information for locating the add-on configuration.
+type ConfigCoordinates struct {
+    // Deprecated: Use configGVR filed instead
+    // crdName is the name of the CRD used to configure instances of the managed add-on.
+    // This field should be configured if the add-on have a CRD that controls the configuration of the add-on.
+    // +optional
+    CRDName string `json:"crdName,omitempty"`
+
+    // Deprecated: Use configGVR filed instead
+    // crName is the name of the CR used to configure instances of the managed add-on.
+    // This field should be configured if add-on CR have a consistent name across the all of the ManagedCluster
+    // instaces.
+    // +optional
+    CRName string `json:"crName,omitempty"`
+
+    // configGVR represents the GroupVersionResource of the add-on configuration.
+    // +optional
+    ConfigGVR ConfigGVR `json:"configGVR,omitempty"`
+    
+    // defaultConfig represents the namespace and name of the default add-on configuration.
+    // In scenario where all add-ons have a same configuration.
+    // +optional
+    DefaultConfig ConfigReferent `json:"defaultConfig,omitempty"`
+}
+
+// ConfigGVR represents the GroupVersionResource of the add-on configuration
+type ConfigGVR struct {
+    // group of the add-on configuration.
+    Group string `json:"group"`
+
+    // version of the add-on configuration.
+    Version string `json:"version"`
+
+    // resource of the add-on configuration.
+    Resource string `json:"resource"`
+}
+
+// ConfigReferent represents the namespace and name for an add-on configuration.
+type ConfigReferent struct {
+    // namespace of the add-on configuration.
+    // If this field is not set, the configuration is in the cluster scope.
+    // +optional
+    Namespace string `json:"namespace,omitempty"`
+
+    // name of the add-on configuration.
+    Name string `json:"name"`
+}
+```
+
+When the `ConfigCoordinates` is defined in the `ClusterManagementAddOnSpec`, the addon-framework will sync the
```

Comment:
How will the addon know when the configuration has been updated?

Reply:
in my mind, framework will watch the configuration resource and update addon status (lastObservedGeneration), which then trigger the manifests interface which addon implement. 

The addon then needs to read this configuration itself and render the agent's manifests. @skeeey i am not sure if framework's higher level interface would look like. This might also need to be added in this proposal.